sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala

```diff
@@ -101,6 +99,9 @@ private[spark] object CosmosConfigNames {
   val WriteBulkUpdateColumnConfigs = "spark.cosmos.write.bulkUpdate.columnConfigs"
   val WriteStrategy = "spark.cosmos.write.strategy"
   val WriteMaxRetryCount = "spark.cosmos.write.maxRetryCount"
+  val WriteFlushCloseIntervalInSeconds = "spark.cosmos.write.flush.intervalInSeconds"
+  val WriteMaxNoProgressIntervalInSeconds = "spark.cosmos.write.flush.noProgress.maxIntervalInSeconds"
+  val WriteMaxRetryNoProgressIntervalInSeconds = "spark.cosmos.write.flush.noProgress.maxRetryIntervalInSeconds"
```

Comment:
I am wondering exposing these as public config vs internal implementation details. How customer would reason about the values to be configured here?

Reply:
I intentionally did not add these settings in the public documentation. We have few other undocumented configs. But these could become handy in few edge cases when you would want to change the retry interval/number